### DISCUSSION TITLE
Views 2.0.0: Advanced column chart styling (gradient bars, value formatter, bar width)

### DISCUSSION THREAD

**Discussion by kroegerama** [patrykandpatrick/vico#675] [Questions] [answered]
Hi there,
I am currently evaluating this library and have a few questions. I'm using the **view**-library version `2.0.0-alpha.17` and want to display a bar (column) chart.

1. Is there a way to fill the bars with a linear gradient (top to bottom)?
2. How can I attach a value formatter after defining styles via styles.xml?
  `barChart.chart!!.bottomAxis = HorizontalAxis.build { ... }` resets the whole style. Is there a `bottomAxis.newBuilder()` or something like that?
3. ~~How can I make the bars wider? I only have one column in the chart and the bars are less than a third as wide as the actual column.~~

Congrats on this great library, looks really good so far!

**Comment by kroegerama** [comment]
I was able to find the solution for 3. myself: The style attribute `ColumnLayerStyle.columnOuterSpacing`.

**Reply by patrickmichalik** [reply]
I’m glad to hear that!

**Comment by patrickmichalik** [comment, marked as answer]
Hello, and thanks for the kind words!

1. You can do this via `LineComponent#dynamicShader`. A `ColumnCartesianLayer.ColumnProvider` will have to be created programmatically. Assuming your `CartesianChart` consists of a single `ColumnCartesianLayer` with one column `LineComponent`, something along these lines can be used:

    ```kt
    (barChart.chart?.layers?.get(0) as ColumnCartesianLayer).columnProvider =
        ColumnCartesianLayer.ColumnProvider.series(
            LineComponent(color = ..., dynamicShader = DynamicShader.verticalGradient(...), ...),
        )
    ```

2. There’s no need to create a new `HorizontalAxis`—this should do the trick:

    ```kt
    (barChart.chart?.bottomAxis as BaseAxis).valueFormatter = ...
    ```

If you have any questions, let me know.
